pkg/controller/controller_binding.go

```diff
@@ -847,25 +845,21 @@ func (c *controller) injectServiceBinding(binding *v1beta1.ServiceBinding, crede
 		// Update existing secret
 		if !IsControlledBy(existingSecret, binding) {
 			controllerRef := GetControllerOf(existingSecret)
-			// TODO mark as terminal error once we have the terminal condition
-			return fmt.Errorf("Secret '%s' is not owned by ServiceBinding, controllerRef: %v", existingSecret.Name, controllerRef)
+			return fmt.Errorf(`Secret "%s/%s" is not owned by ServiceBinding, controllerRef: %v`, binding.Namespace, existingSecret.Name, controllerRef)
```

Comment:
not owned by ServiceBinding
or not owned by ServiceCatalog?

Reply:
ServiceBinding is the right thing to do here - the message eventually logged will say which binding is supposed to own the secret.

Reply:
Not to be tense, but is there never a case where a Secret could be created outside of Service Catalog that could clash with the names? I thought that ControllerOf was optional field that we filled in. That's the case I was worried about, but perhaps I'm confused here.

Reply:
Yeah, there are a couple of cases at least:

- Another piece of software that sets controller ref could have created the secret we want to manage, and set the controller ref accordingly (or not)
- A user could have mistakenly specified the same secret name on multiple bindings, and then there's a race condition
- A user could have created the secret themselves for a number of other cases

I do think now that we should specify which binding we want to control the secret in this log message - since you could have two catalog resources racing.  Did I answer your question otherwise?

Reply:
Yes, my point was that the controller is the one that's marked as owning, not the binding. and that there can be cases where the secret is not owned by a binding nor the service catalog.

Reply:
I think in this case, the ServiceBinding is the 'controller' that results in the secret being created.  I don't think the field is especially well named, but that's what the deal is.